source/vibe/inet/url.d

```diff
@@ -92,11 +92,31 @@ struct URL {
 					}
 
 					m_host = str[hs .. si];
-					auto pi = m_host.indexOfCT(':');
-					if(pi > 0) {
-						enforce(pi < m_host.length-1, "Empty port in URL.");
-						m_port = to!ushort(m_host[pi+1..$]);
-						m_host = m_host[0 .. pi];
+
+					long findPort ( string src )
```

Comment:
This must be either `sizediff_t` or `auto` for the slice operation below to compile with 32-bits.